* Allows control of the audio mixer even for media with no audio content
  (e.g. background.avi or tv:// with immediatemode=1). Special case handling
  is made for dvd's and .ifo files, presumably because different sections of
  the media may not have audio, but when going from chapter to chapter if the
  audio has already been initialized there would be an issue.
--- old/command.c	2006-05-27 02:33:57.000000000 +0800
+++ new/command.c	2006-05-27 02:34:09.000000000 +0800
@@ -409,8 +409,10 @@
 			      MPContext * mpctx)
 {
 
+#if 0
     if (!mpctx->sh_audio)
 	return M_PROPERTY_UNAVAILABLE;
+#endif
 
     switch (action) {
     case M_PROPERTY_GET:
@@ -465,8 +467,10 @@
 			    MPContext * mpctx)
 {
 
+#if 0
     if (!mpctx->sh_audio)
 	return M_PROPERTY_UNAVAILABLE;
+#endif
 
     switch (action) {
     case M_PROPERTY_SET:
--- old/mplayer.c	2006-05-27 02:33:57.000000000 +0800
+++ new/mplayer.c	2006-05-27 02:34:09.000000000 +0800
@@ -1542,6 +1542,20 @@
   mpctx->mixer.audio_out = mpctx->audio_out;
   mpctx->mixer.volstep = volstep;
 }
+else if (!strncmp (filename, "dvd", 3) || !strncmp (filename, "vcd", 3) || strcasestr (filename, ".ifo")) {
+  mpctx->audio_out=NULL;
+}
+else{
+#if WORDS_BIGENDIAN
+    if(mpctx->audio_out=init_best_audio_out(audio_driver_list,0,11025,2,AF_FORMAT_S16_BE,0)){
+#else
+    if(mpctx->audio_out=init_best_audio_out(audio_driver_list,0,11025,2,AF_FORMAT_S16_LE,0)){
+#endif
+        inited_flags|=INITED_AO;
+        mpctx->mixer.audio_out = mpctx->audio_out;
+        mpctx->mixer.volstep = volstep;
+    }
+}
 }
 
 
